plugin/client.go

```diff
@@ -0,0 +1,128 @@
+package plugin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/cloudquery/plugin-sdk/v4/message"
+	"github.com/cloudquery/plugin-sdk/v4/plugin"
+	"github.com/cloudquery/plugin-sdk/v4/scheduler"
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+	"github.com/cloudquery/plugin-sdk/v4/state"
+	"github.com/cloudquery/plugin-sdk/v4/transformers"
+	"github.com/coinpaprika/cq-source-coinpaprika/client"
+	"github.com/coinpaprika/cq-source-coinpaprika/resources/services/coins"
+	"github.com/coinpaprika/cq-source-coinpaprika/resources/services/exchanges"
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const maxMsgSize = 100 * 1024 * 1024 // 100 MiB
+
+type Client struct {
+	logger      zerolog.Logger
+	config      client.Spec
+	backendConn *grpc.ClientConn
+	scheduler   *scheduler.Scheduler
+	tables      schema.Tables
+	plugin.UnimplementedDestination
+}
+
+func (c *Client) Close(_ context.Context) error {
+	if c.backendConn != nil {
+		return c.backendConn.Close()
+	}
+	return nil
+}
+
+func (c *Client) Tables(_ context.Context, options plugin.TableOptions) (schema.Tables, error) {
+	tt, err := c.tables.FilterDfs(options.Tables, options.SkipTables, options.SkipDependentTables)
+	if err != nil {
+		return nil, err
+	}
+	return tt, nil
+}
+
+func (c *Client) Sync(ctx context.Context, options plugin.SyncOptions, res chan<- message.SyncMessage) error {
+	tt, err := c.tables.FilterDfs(options.Tables, options.SkipTables, options.SkipDependentTables)
```

Comment:
verify for no connection, see https://github.com/cloudquery/cloudquery/blob/main/plugins/source/gcp/resources/plugin/client.go